commands/assets/add.go

```diff
@@ -0,0 +1,135 @@
+package assets
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+
+	"github.com/netlify/netlifyctl/commands/middleware"
+	"github.com/netlify/netlifyctl/context"
+	"github.com/netlify/open-api/go/plumbing/operations"
+	"github.com/spf13/cobra"
+)
+
+type assetsAddCmd struct {
+	private bool
+}
+
+func setupAddCommand(middlewares []middleware.Middleware) *cobra.Command {
+	cmd := &assetsAddCmd{}
+	ccmd := &cobra.Command{
+		Use:   "add [ASSET PATH 1] [ASSET PATH 2] ...",
+		Short: "Add an asset to a site",
+		Long:  "Add an asset to a site",
+	}
+	ccmd.Flags().BoolVarP(&cmd.private, "private", "p", false, "make the asset private")
+
+	return middleware.SetupCommand(ccmd, cmd.AddAsset, middlewares)
+}
+
+func (c *assetsAddCmd) AddAsset(ctx context.Context, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("missing asset paths to upload")
+	}
+
+	siteId, err := siteIdForCommand(cmd)
+	if err != nil {
+		return err
+	}
+
+	dups := make(map[string]os.FileInfo)
+	for _, arg := range args {
+		fi, err := os.Stat(arg)
+		if err != nil {
+			return fmt.Errorf("unable to access asset with path %s, all assets must exist: %v", arg, err)
+		}
+
+		if _, dup := dups[arg]; dup {
+			fmt.Printf("[WARNING] duplicated asset %s, will be uploaded only once\n", arg)
+		}
+
+		dups[arg] = fi
+	}
+
+	client := context.GetClient(ctx)
+	params := operations.NewCreateSiteAssetParams().WithSiteID(siteId)
+	if c.private {
+		visibility := "private"
+		params = params.WithVisibility(&visibility)
+	}
+
+	for fp, fi := range dups {
+		body, err := os.Open(fp)
+		if err != nil {
+			return err
+		}
+		defer body.Close()
+
+		buffer := make([]byte, 512)
+		n, err := body.Read(buffer)
+		if err != nil && err != io.EOF {
+			return err
+		}
+		contentType := http.DetectContentType(buffer[:n])
+
+		params = params.WithName(fi.Name()).WithSize(fi.Size()).WithContentType(contentType)
+
+		signature, err := client.AddSiteAsset(ctx, params)
+		if err != nil {
+			return err
+		}
+
+		body.Seek(0, 0)
+		bufferWriter := &bytes.Buffer{}
+		writer := multipart.NewWriter(bufferWriter)
+
+		for key, value := range signature.Form.Fields {
+			writer.WriteField(key, value)
+		}
+
+		part, err := writer.CreateFormFile("file", fi.Name())
+		if err != nil {
+			return err
+		}
+
+		if _, err = io.Copy(part, body); err != nil {
+			return err
+		}
+
+		err = writer.Close()
+		if err != nil {
+			return err
+		}
+
+		req, err := http.NewRequest("POST", signature.Form.URL, bufferWriter)
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Content-Type", writer.FormDataContentType())
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode != 201 {
+			defer resp.Body.Close()
+			respBody, _ := ioutil.ReadAll(resp.Body)
```

Comment:
why are you ignoring the error here? It will let us know about things like hangups from the system. And I don't think respBody is safe to use if there is an error.